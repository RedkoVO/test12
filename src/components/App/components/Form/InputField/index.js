import React from 'react'
import cn from 'classnames'

const InputField = ({ input, type, placeholder}) => {

  return (
    <input
      {...input}
      type={type}
      placeholder={placeholder}
    />
  )
}

export default InputField