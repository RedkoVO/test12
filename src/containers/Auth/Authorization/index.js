import compose from 'recompose/compose'
import { connect } from 'react-redux'
import { withHandlers, withState, pure } from 'recompose'
import { reduxForm } from 'redux-form'
import validate from './validate'
import BigNumber from 'bignumber.js'

import { sequential } from '../../../utils/promiseSequential'

import withConfigAndAllBalance from '../../../hocs/withConfigAndAllBalance'

import { getAllBalanceInfo, getIncoming } from '../../../redux/actions/balance'
import { getAllBalanceInfoSelector } from '../../../selectors/balance'
import { getConfigSelector } from '../../../selectors/config'
import { receiveFromFaucet } from '../../../requests/receiveFromFaucet'
import Crypto from '../../../crypto/crypto'

import AsyncAuthorizationDesktop from '../../../components/Auth/Authorization/Desktop/AsyncAuthorizationDesktop'

const FORM_NAME = 'registration'

const mapStateToProps = state => ({
  allBalance: getAllBalanceInfoSelector(state),
  config: getConfigSelector(state)
})

export default compose(
  connect(mapStateToProps),
  withConfigAndAllBalance,
  reduxForm({
    form: FORM_NAME,
    validate
  }),
  withState('isDisabledButton', 'setDisabledButton', false),
  withHandlers({
    onSubmit: ({
      handleSubmit,
      history,
      dispatch,
      isDisabledButton,
      setDisabledButton,
      allBalance
    }) =>
      handleSubmit(variables => {
        const getCryptoInfo = Crypto.account.accountFromSecret(variables.key)
        localStorage.setItem('secretKey', getCryptoInfo.secretKey)
        localStorage.setItem('address', getCryptoInfo.address)
        localStorage.setItem('publicKey', getCryptoInfo.publicKey)

        /* TODO: refactor !!! */
        if (!isDisabledButton) {
          setDisabledButton(!isDisabledButton)

          const data = {
            address: localStorage.getItem('address')
          }

          dispatch(getAllBalanceInfo(data))
            .then(res => {
              if (res.success) {
                const dataForIncoming = {
                  address: localStorage.getItem('address')
                }
                dispatch(getIncoming(dataForIncoming))
                  .then(blocks => {
                    const requestsArr = []
                    let userAccount = {
                      publicKey: localStorage.getItem('publicKey'),
                      secretKey: localStorage.getItem('secretKey'),
                      address: localStorage.getItem('address'),
                      representative: localStorage.getItem('representative'),
                      lastBlock: localStorage.getItem('lastBlock'),
                      balance: new BigNumber(res.balance)
                    }

                    // Object.keys(blocks.result).forEach(hash => {
                    //   console.log('hash', hash)
                    //   const sourceBlockHash = hash
                    //   const amountStr = blocks.result[hash].amount

                    //   requestsArr.push(() =>
                    //     receiveFromFaucet(
                    //       userAccount,
                    //       sourceBlockHash,
                    //       amountStr
                    //     )
                    //   )
                    // })

                    console.log('res', res)

                    blocks.result.forEach(item => {
                      console.log('item', item)
                      // console.log('-----', res.)
                      const sourceBlockHash = item.hash
                      const amountStr = item.amount

                      requestsArr.push(() =>
                        receiveFromFaucet(
                          userAccount,
                          sourceBlockHash,
                          amountStr
                        )
                      )
                    })

                    sequential(requestsArr).then(() => {
                      dispatch(getAllBalanceInfo(data))
                        .then(res => {
                          history.push('/')

                          if (res && res.lastBlock) {
                            localStorage.setItem('lastBlock', res.lastBlock)
                          }
                        })
                        .catch(err => {
                          console.log(
                            'ERROR getAllBalanceInfo inside incoming',
                            err
                          )
                        })
                    })
                  })
                  .catch(err => {
                    console.log('Error get incoming', err)
                  })
              }
            })
            .catch(err => {
              console.log('Error get balance', err)
            })
        }
      })
  }),
  pure
)(AsyncAuthorizationDesktop)
