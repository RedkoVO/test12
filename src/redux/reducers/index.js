import { combineReducers } from 'redux'
import { reducer as form } from 'redux-form'
import auth from './auth'
import balance from './balance'

export default combineReducers({
  form,
  auth,
  balance
})