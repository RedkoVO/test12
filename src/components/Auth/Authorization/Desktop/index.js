import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import withStyles from '@material-ui/core/styles/withStyles'
import { Field, Form } from 'redux-form'
import cn from 'classnames'

import InputField from '../../../App/components/Form/InputField'

import styles from './styles'

const Authorization = ({ classes, onSubmit, isDisabledButton }) => (
  <div className={classes.root}>
    <div className={classes.containerRegistrat}>
      <div className={classes.containerRegistratStep1}>
        <Form className={classes.registrStep1} onSubmit={onSubmit}>
          <h1 className={classes.step1Title}>Login</h1>
          <label htmlFor="key" className={classes.step1Label}>
            Enter secret key from email
          </label>
          <Field
            id="key"
            name="key"
            type="text"
            className={classes.field}
            component={InputField}
            placeholder="Secret key"
          />

          <div>
            <Link to={'/registration'} className={classes.step1BtnBack}>
              Registration
            </Link>
            <button
              type="submit"
              className={cn(classes.step1BtnsRegister, {
                disabled: isDisabledButton
              })}
            >
              Login!
            </button>
          </div>
        </Form>
      </div>

      <div className={classes.footer}>
        <span className={classes.copyright}>
          Vault. All Rights Reserved 2018
        </span>
        <div>
          <a href="/" className={classes.footerLink}>
            Privacy Policy
          </a>
          <a href="/" className={classes.footerLink}>
            Terms and conditions
          </a>
        </div>
      </div>
    </div>
  </div>
)

Authorization.propTypes = {
  classNamees: PropTypes.object,
  onSubmit: PropTypes.func,
  isDisabledButton: PropTypes.bool,
  test_name: PropTypes.string
}

export default withStyles(styles)(Authorization)
