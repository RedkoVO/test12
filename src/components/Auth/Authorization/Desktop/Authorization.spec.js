import * as React from 'react'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { reduxForm, Field } from 'redux-form'
import { createStore } from 'redux'

import InputField from '../../../App/components/Form/InputField'
import PageLayout from '../../../App/components/PageLayout'

import Authorization from './'

const Wrapper = reduxForm({ form: 'test' })(({ children }) => children)
const store = createStore(a => a, {})

describe('<Authorization />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter keyLength={0}>
            <Wrapper>
              <Authorization onSubmit={() => {}} isDisabledButton={false} />
            </Wrapper>
          </MemoryRouter>
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('consist InputField component', () => {
    const tree = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={[ '/registration' ]}>
          <Wrapper>
            <Authorization onSubmit={() => {}} isDisabledButton={false} />
          </Wrapper>
        </MemoryRouter>
      </Provider>
    )

    const input = tree.find('input#key')
    input.simulate('change', {target: {value: '1212TESTTEST'}})

    const form = tree.find('form').first()
    form.simulate('submit')

    tree.update()

    console.log('----------', input.debug())
    // console.log('----------', tree.find(InputField).debug())

    expect(input.prop('value')).toBe('1212TESTTEST')
    expect(tree.find(InputField)).toHaveLength(1)
  })



  it('HTML: check input #key element', () => {
    const tree = shallow(
      <Authorization onSubmit={() => {}} isDisabledButton={false} />
    )

    // console.log('----------', tree.dive().find('#key').debug())

    expect(tree.dive().find('#key').exists()).toEqual(true)
  })
})
