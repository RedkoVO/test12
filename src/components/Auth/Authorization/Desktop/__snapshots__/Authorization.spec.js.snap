// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authorization /> renders correctly 1`] = `
<div
  className="Authorization-root-1"
>
  <div
    className="Authorization-containerRegistrat-3"
  >
    <div
      className="Authorization-containerRegistratStep1-2"
    >
      <form
        className="Authorization-registrStep1-7"
        onSubmit={[Function]}
      >
        <h1
          className="Authorization-step1Title-8"
        >
          Login
        </h1>
        <label
          className="Authorization-step1Label-9"
          htmlFor="key"
        >
          Enter secret key from email
        </label>
        <input
          className="Authorization-field-10"
          id="key"
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="Secret key"
          type="text"
          value=""
        />
        <div>
          <a
            className="Authorization-step1BtnBack-11"
            href="/registration"
            onClick={[Function]}
          >
            Registration
          </a>
          <button
            className="Authorization-step1BtnsRegister-12"
            type="submit"
          >
            Login!
          </button>
        </div>
      </form>
    </div>
    <div
      className="Authorization-footer-4"
    >
      <span
        className="Authorization-copyright-5"
      >
        Vault. All Rights Reserved 2018
      </span>
      <div>
        <a
          className="Authorization-footerLink-6"
          href="/"
        >
          Privacy Policy
        </a>
        <a
          className="Authorization-footerLink-6"
          href="/"
        >
          Terms and conditions
        </a>
      </div>
    </div>
  </div>
</div>
`;
